First Ansible Roles API test

sambible opened the pull request:
Requires https://github.com/SatelliteQE/nailgun/pull/841

First test surrounding Ansible Roles

adarshdubey-star reviewed (changes requested):
Please add :expectedresults: as well

Gauravtalreja1 commented on tests/foreman/api/test_ansible.py:
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
+
+    :id: 95d0213e-a5a9-4671-8a9e-89a77fc67f39
+
+    :customerscenario: false
```
:customerscenario: is only required when it is true, please remove

Gauravtalreja1 commented on tests/foreman/api/test_ansible.py:
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
+    :customerscenario: false
+
+    :steps:
+
```
```suggestion
```

Griffin-Sullivan commented on tests/foreman/api/test_ansible.py:
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
+
+    :CaseAutomation: Automated
+    """
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    host = target_sat.api.Host().create()
+    id = target_sat.api.SmartProxy(name=target_sat.hostname).search()[0].id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+    host.assign_roles(data={'ansible_role_ids': [1]})
+    host_roles = host.list_roles()
+    assert host_roles[0]['name'] == SELECTED_ROLE
```
If we have a list of all roles, we could turn most of this into a fixture and parametrize the test with the list of roles.

sambible replied:
Hmm, I do like that idea
This is just the first of the available roles on any given satellite. Maybe make a constant list of those, and then do as you suggest

Griffin-Sullivan commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

Griffin-Sullivan commented on tests/foreman/api/test_ansible.py:
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
+    host = target_sat.api.Host().create()
+    id = target_sat.api.SmartProxy(name=target_sat.hostname).search()[0].id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+    host.assign_roles(data={'ansible_role_ids': [1]})
```
```suggestion
    host.assign_role(data={'ansible_role_ids': [1]})
```

sambible replied:
ah, yeah had to update my nailgun pr - changed the name to be consistent with the behavior

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

Gauravtalreja1 commented on tests/foreman/api/test_ansible.py:
```diff
@@ -59,3 +59,26 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
+
+    :id: 95d0213e-a5a9-4671-8a9e-89a77fc67f39
+
+    :steps:
+        1. Create a Host
+        2. Import a roles into satellite
+        3. Assign that role to a host
+        4. Assert that the role was assigned to the host successfully
```
Ack, only pending @adarshdubey-star 's comment

Gauravtalreja1 replied:
@sambible this is pending since 2 months, could you pls address it?

ogajduse commented on tests/foreman/api/test_ansible.py:
```diff
@@ -59,3 +59,26 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
+    """
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    host = target_sat.api.Host().create()
+    id = target_sat.api.SmartProxy(name=target_sat.hostname).search()[0].id
```
```suggestion
    id = target_sat.internal_capsule.id
```

ogajduse replied:
@sambible Oops, apologies. It should be either `nailgun_capsule`, or `nailgun_smart_proxy`.

adarshdubey-star commented:
@sambible any updates on this PR?

Gauravtalreja1 commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

sambible commented:
Addressed comments and implemented @ogajduse change request. Going to run the test again with runner, but should be good to go when we can merge the Nailgun PR

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

adarshdubey-star commented on tests/foreman/api/test_ansible.py:
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
+    host = target_sat.api.Host().create()
+    id = target_sat.internal_capsule.id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+    host.assign_roles(data={'ansible_role_ids': [1]})
```
```suggestion
    host.assign_ansible_roles(data={'ansible_role_ids': [1]})
```

adarshdubey-star replied:
I think this is the cause of the PRT failure seeing you nailgun PR

adarshdubey-star commented on tests/foreman/api/test_ansible.py:
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
+    id = target_sat.internal_capsule.id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+    host.assign_roles(data={'ansible_role_ids': [1]})
+    host_roles = host.list_roles()
```
```suggestion
    host_roles = host.list_ansible_roles()
```

adarshdubey-star commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

devendra104 commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

omkarkhatavkar commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

sambible commented:
Nailgun PR is merged, this can go In now.

Griffin-Sullivan commented:
@sambible rebase and we can merge!

Griffin-Sullivan commented:
I guess we need to rebase again

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'

JacobCallahan commented:
@sambible Can you check on the PRT failure and revisit this PR? It would be nice to have this merged.

Griffin-Sullivan commented:
I vote we just close this PR as [tests/foreman/api/test_ansible.py::test_positive_ansible_job_on_host](https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/api/test_ansible.py#L70) has already covered what this test is doing. There are also equivalent tests in CLI and UI now, so I don't see the need to keep this test when it is already set up for those other tests.
